config/src/main/java/org/springframework/security/config/web/util/matcher/ServletRequestMatcherBuilderFactory.java

```diff
@@ -0,0 +1,162 @@
+ * A factory for constructing {@link RequestMatcherBuilder} instances based on the type of
+ * servlet. If the servlet is {@link org.springframework.web.servlet.DispatcherServlet},
+ * then it will return a {@link MvcRequestMatcherBuilder}; otherwise, it will return a
+ * {@link AntPathRequestMatcherBuilder}.
+ *
+ * <p>
+ * NOTE: In many cases, which kind of request matcher builder is needed is apparent by the
+ * servlet configuration, and so you should generally use
+ * {@link RequestMatcherBuilders#createDefault} instead of this class. Use this class when
+ * you want or need to indicate which request matcher URIs belong to which servlet.
+ *
+ * <p>
+ * In all cases, you can indicate the servlet path by using the {@link #servletPath}
+ * method.
+ *
+ * <p>
+ * Consider, for example, the circumstance where you have Spring MVC configured and also
+ * Spring Boot H2 Console. Spring MVC registers a servlet of type
+ * {@link DispatcherServlet} as the default servlet and Spring Boot registers a servlet of
+ * its own as well at `/h2-console/*`.
+ *
+ * <p>
+ * Such might have a configuration like this in Spring Security: <code>
+ * 	http
+ *		.authorizeHttpRequests((authorize) -> authorize
+ *	        .requestMatchers("/js/**", "/css/**").permitAll()
+ *	        .requestMatchers("/my/controller/**").hasAuthority("CONTROLLER")
+ *	        .requestMatchers("/h2-console/**").hasAuthority("H2")
+ *	        .anyRequest().authenticated()
+ *	    )
+ *	    // ...
+ * </code>
+ *
+ * <p>
+ * Spring Security by default addresses the above configuration on its own.
+ *
+ * <p>
+ * However, consider the situation where there is more than one {@link DispatcherServlet},
+ * say one for static resources mapped to `/static/**` and one for controllers mapped to
+ * `/app/*`. In this case, the above configuration is ambiguous, and you should use this
+ * class to clarify the rest of each MVC URI like so: <code>
+ *  ServletRequestMatcherBuilderFactory factory = new ServletRequestMatcherBuilderFactory(context);
+ * 	http
+ *		.authorizeHttpRequests((authorize) -> authorize
+ * 			.requestMatchers(factory.servletPath("/static").matchers("/js/**", "/css/**")).permitAll()
+ *	        .requestMatchers(factory.servletPath("/app").matchers("/my/controller/**")).hasAuthority("CONTROLLER")
+ *	        .requestMatchers(factory.servletPath("/h2-console").matchers("/**")).hasAuthority("OTHER")
+ *	        .anyRequest().authenticated()
+ *	    )
+ *	    // ...
+ * </code>
+ *
+ * <p>
+ * In the above configuration, it's now clear to Spring Security that the following
+ * matchers map to these corresponding URIs:
+ *
+ * <ul>
+ * <li>`/js/**` ==> `/static/js/**`</li>
+ * <li>`/css/**` ==> `/static/css/**`</li>
+ * <li>`/my/controller/**` ==> `/app/my/controller/**`</li>
+ * <li>`/h2-console/**` ==> `/h2-console/**`</li>
+ * </ul>
+ *
+ * @author Josh Cummings
+ * @since 6.2
+ * @see AbstractRequestMatcherRegistry
+ */
+public final class ServletRequestMatcherBuilderFactory {
```

Comment:
I like that this is separate from the `RequestMatcherBuilder` and focused solely on specifying a particular servlet. It helps clarify the difference between the steps needed to get a `RequestMatcher` instance.